tmt/base.py

```diff
@@ -277,6 +284,22 @@ def from_spec(cls, raw: _RawRequireFmfId) -> 'RequireFmfId':  # type: ignore[ove
 Require = Union[str, RequireFmfId]
 
 
+# TODO: there might be a way to do this transparently, with some __str__ magic.
+# A dedicated, "remember-to-call" function is far from ideal.
+def printable_require_item(require: Require) -> Union[str, _RawRequireFmfId]:
+    """
+    Small helper for producing printable version of a requirement.
+
+    A requirement may be a string or :py:class:`RequireFmfId`, the later needs special
+    handling, therefore a small helper to handle details.
+    """
+
+    if isinstance(require, RequireFmfId):
+        return require.to_minimal_dict()
+
+    return require
+
+
 def normalize_require(raw_require: Optional[_RawRequire]) -> List[Require]:
```

Comment:
Now, when both `require` and `recommend` are handled in the same way, calling methods common for both of them with "require" seems a little bit misleading. Perhaps, we could use the general term `dependency`? Not sure if it's worth to rename `RequireFmfId` --> `DependencyFmfId` and all their friends though.

Reply:
Good, let's do the rename in a separate patch then.